### PR TITLE
Call the EngineStateReport via Manager method

### PR DESCRIPTION
**Pull request by lleshchi**
Closes: https://github.com/stratis-storage/stratis-cli/issues/700

### CODE REVIEW

**Comment by lleshchi** [comment]
@mulkieran If this looks good, I will continue on with tests and man pages

**Comment by mulkieran** [comment]
@lleshchi Just explicitly allow the bad-continuation errors. black is correct, pylint is wrong.

File: src/stratis_cli/_actions/_top.py [hunk 1]
```diff
@@ -652,11 +652,24 @@ def get_report(namespace):
         :raises StratisCliEngineError:
         """
         # pylint: disable=import-outside-toplevel
-        from ._data import Report
+        if (
```
Comment (mulkieran) [inline src/stratis_cli/_actions/_top.py:655]: ```report_name``` will always have some non-```None``` value, due to the parser change. So it's not necessary to check for it here.

**Review by mulkieran — CHANGES_REQUESTED** [review]
That's all I see. You should be able to carry on w/ the rest of the steps after fixing those.

**Review by jbaublitz — COMMENTED** [review]
Other than @mulkieran's request, this looks good to me so far!

**Comment by lleshchi** [comment]
@mulkieran The `get_report()` and `get_engine_state_report()` methods in testlib/dbus.py can be condensed into one method in order to satisfy the pylint error "too-many-public-methods" or error can be ignored

**Comment by mulkieran** [comment]
> @mulkieran The `get_report()` and `get_engine_state_report()` methods in testlib/dbus.py can be condensed into one method in order to satisfy the pylint error "too-many-public-methods" or error can be ignored

@lleshchi: I'm fine if you disable the warning. I think the code is simpler as it is.

File: src/stratis_cli/_parser/_parser.py [hunk 2]
```diff
@@ -142,7 +142,10 @@ def wrapped_func(*args):
                 (
                     "report_name",
                     dict(
-                        default=None, type=str, help=("Name of the report to display")
+                        default="engine_state_report",
+                        type=str,
+                        help=("Name of the report to display"),
+                        nargs="?",
```
Comment (mulkieran) [inline src/stratis_cli/_parser/_parser.py:148]: Please add a `choices` parameter here as well with the two available names, ```engine_state_report``` and ```errored_pool_report```.

File: docs/stratis.txt [hunk 3]
```diff
@@ -96,9 +96,10 @@ key unset <key_desc>::
      Unset a key in the kernel keyring so it is no longer available for encryption
      operations.
 report <report_name>::
-        Get a report from the daemon regarding its internal state. This command is
-        not supported as the report names and returned JSON schemas are unstable
-        and may be changed at any time.
+        Get a report from the daemon regarding its internal state.
+        Get engine_state_report if no report name is provided. This command only
```
Comment (mulkieran) [inline docs/stratis.txt:100]: What we want here is to indicate which names are stable and which aren't, but making a list of 1 is unappealing.

Maybe say something like:

The `engine_state_report` name will be supported in future releases. Any other report name should be considered unstable and may be removed in a future release. The JSON schema of any report must always be considered unstable.

**Review by mulkieran — CHANGES_REQUESTED** [review]
Two final requests...otherwise it looks ready.

**Review by mulkieran — CHANGES_REQUESTED** [review]
Looks good! Please close this and open a new PR for final review.

**Pull request closed without merging**